### PR TITLE
Arm backend: add SmolLM2 Ethos-U export, generation and eval flow

### PR DESCRIPTION
**Pull request by xingguo01**
- semihosting and FVP runner build helpers
- sampled text generation from prompt files
- Wikitext full-logits perplexity evaluation on FVP
- example prompts and documentation for reproducing results

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell @rascani

### CODE REVIEW

File: examples/arm/smollm2_example_ethos_u/README.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+Install the Arm Ethos-U dependencies and generate `setup_path.sh`:
+
+```bash
+examples/arm/setup.sh \
+  --i-agree-to-the-contained-eula \
+  --enable-ethos-u-deps
+```
+
+Activate your Python environment and source the generated Arm setup:
```
Comment (zingo) [inline examples/arm/smollm2_example_ethos_u/README.md]: I think the env setup needs to be before examples/arm/setup.sh as it installs stuff
As the general setup mention conda maybe we should write something like " with conda or venv" 
Or just mention it in text and point to this doc https://docs.pytorch.org/executorch/stable/using-executorch-building-from-source.html ? (but with a  relative path)

**Review by zingo — CHANGES_REQUESTED** [review]

File: examples/arm/smollm2_example_ethos_u/README.md (hunk 1 above)
Comment (xingguo01) [reply examples/arm/smollm2_example_ethos_u/README.md]: Good point, updated the prerequisites so the conda/venv environment is activated before running examples/arm/setup.sh, since setup.sh installs Python packages into the active environment. I also linked to the local ExecuTorch build-from-source doc for the general environment setup. It should be noted that I was mimicking the image classification ethos u demo

**Review by zingo — APPROVED** [review]
